Add sorting via TLvlGraph.

User4martin opened the pull request:
Only node positions are taken, routing of edges is not affected.

Maybe useful. Please review.

Commit:
Add sorting via TLvlGraph. Only node positions are taken, routing of edges is not affected.

PascalCorpsman commented on unit1.lfm:
```diff
@@ -148,6 +148,10 @@ object Form1: TForm1
       OnClick = MenuItem10Click
     end
+    object MenuItem48: TMenuItem
+      Caption = 'Arrange all nodes by dependency'
```
Caption = 'Arrange all visible nodes by dependency'

PascalCorpsman commented on unit1.pas:
```diff
@@ -482,6 +484,63 @@
+
+  LvlGrph := TLvlGraph.create;
+  For i := 0 To GraphBox1.Graph.NodeCount - 1 Do
+    LvlGrph.GetNode(GraphBox1.Graph.Node[i].Caption, True);
```
If GraphBox1.Graph.Node[i].Visible Then Begin
      LvlGrph.GetNode(GraphBox1.Graph.Node[i].Caption, True);
    End;

PascalCorpsman commented on unit1.pas:
```diff
@@ -482,6 +484,63 @@
+        GraphBox1.Graph.Node[GraphBox1.Graph.Node[i].Edges[j].StartIndex].Caption,
+        GraphBox1.Graph.Node[GraphBox1.Graph.Node[i].Edges[j].EndIndex].Caption,
+        True
+      );
```
For j := 0 To high(GraphBox1.Graph.Node[i].Edges) Do Begin
      If GraphBox1.Graph.Node[GraphBox1.Graph.Node[i].Edges[j].StartIndex].Visible And
        GraphBox1.Graph.Node[GraphBox1.Graph.Node[i].Edges[j].EndIndex].Visible Then Begin
        LvlGrph.GetEdge(
          GraphBox1.Graph.Node[GraphBox1.Graph.Node[i].Edges[j].StartIndex].Caption,
          GraphBox1.Graph.Node[GraphBox1.Graph.Node[i].Edges[j].EndIndex].Caption,
          True
          );
      End;

PascalCorpsman reviewed (changes requested):
Hello Martin,
i like what you proposed, but i would prefer limiting the "reordering" to the visible nodes.
I added the needed changes into my review comments for you to "add". 
All other changes (code formatting, missing "Begin" "end"..) i will change after i merged your suggestions.

Commit:
Only sort visible nodes

Commit:
Caption

User4martin commented:
Added the `IF`  (3 times, also for the last loop).

Refactored one loop into a nested function.

Done some of the formatting.

----
Will be away, the rest of the week. Online, but not able to make amends to the code.
